### DISCUSSION TITLE
Does this work with the Uno R4 Wifi?

### DISCUSSION THREAD

**Discussion by Binary-System** [Makuna/DFMiniMp3#166] [General]
I used this lib already and on the older project its running ok. Now I'm working with an Uno R4 Wifi and run into unsolveable problems.

I think the player is an original DFRobot one.
Backside chips are lettered with: DFROBOTLISP3 and YXB002D 02109032

I have three of them here

1st Problem: Don't works at all. All I get from the PlayRadom test script is configured with sofware serial on pins 10 and 11: 

`01:52:04.922 -> OUT  7e ff 06 0c 01 00 00 fe ee ef`

Nothing else

2nd Problem: it is running hot within minutes to too hot to touch.

I tried all three of them and I controlled the wires a dozen times, they are like in the example under https://wiki.dfrobot.com/DFPlayer_Mini_SKU_DFR0299#target_5

Is this compatible with the R4 ?

**Comment by Makuna** [comment]
If you never get a debug line starting with "IN 7e..." then the module is not responding to commands.  It should at least reply with an error.  

Did you correctly include the resistor on the TX pin as outlined by DFRobot?
Make sure you power your MP3 board with the correct voltage.

The library should be fully compatible.  The question is the implementation of Serial they provide for the board compatible with Arduino standards.  I would suspect it is though.

**Reply by Binary-System** [reply]
Took a bit because I ordered another R4 and another Player for testing, the other was quite heavily built into stuff already.

Resistor is in place. Generally I followed the setup from

[https://wiki.dfrobot.com/DFPlayer_Mini_SKU_DFR0299#target_5](https://wiki.dfrobot.com/DFPlayer_Mini_SKU_DFR0299#target_5)

I used the example sketch "play random" and switched it to SoftwareSerial and Debug
```

#define DfMiniMp3Debug Serial
// this example will play a random track from all on the sd 
//
// it expects the sd card to contain some mp3 files
#include <SoftwareSerial.h>
#include <DFMiniMp3.h>

// forward declare the notify class, just the name
//
class Mp3Notify; 

// define a handy type using serial and our notify class
//
//typedef DFMiniMp3<HardwareSerial, Mp3Notify> DfMp3; 

// instance a DfMp3 object, 
//
//DfMp3 dfmp3(Serial1);

// Some arduino boards only have one hardware serial port, so a software serial port is needed instead.
// comment out the above definitions and use these
SoftwareSerial secondarySerial(11, 10); // RX, TX
typedef DFMiniMp3<SoftwareSerial, Mp3Notify> DfMp3;
DfMp3 dfmp3(secondarySerial);
```
No changes below the line.

The wiring for testing at the moment:
![image](https://github.com/Makuna/DFMiniMp3/assets/57272041/ce23291e-7549-4ebb-b6c1-33de5837c4cf)

Output in the serial monitor is

```
22:39:04.581 -> initializing...
22:39:04.581 -> OUT  7e ff 06 0c 01 00 00 fe ee ef
22:39:05.502 -> OUT  7e ff 06 0c 01 00 00 fe ee ef
22:39:06.449 -> OUT  7e ff 06 0c 01 00 00 fe ee ef
```

**Reply by Binary-System** [reply]
I searched a bit around and it seems Hardware Serial (Serial1) and SoftwareSerial are both broken on the Uno R4 WiFi. Some have luck on the R4 with specific pin combinations, most success I have found was SoftwareSerial on 0/1, 6/7 and 3/4.

Wasn't a problem of the lib, but the hardware and the Uno libs.
